### ISSUE TITLE
Seperate default_dec_generate versions for 1d and 2d.

### ISSUE THREAD

**Issue by GeorgeR227** [AlgebraicJulia/Decapodes.jl#147]
Since there exist DEC operators that depend on the dimension of the space they are applied on, such as the Lie Derivative and the Interior Product, we should organize these dimension specific implementations by letting the user choose separate default_dec_generate versions for 1d and 2d. Here's an example for 1d:

```julia

function default_dec_generate_1d(...)
    op = @match my_symbol begin
        :L => lie_derivative_1d(...)
        :i => interior_product_1d(...)
        ...
        # For any operators that don't depend on dimensions like d, or hodge star
        _ => default_dec_generate(...)
    end
...

```

While we could do the same by passing a dimension parameter into the current default_dec_generate and then having the Lie Derivative and Interior Product implementations accept that dimension information, I feel this method organizes the operators a bit better for the user.

**Comment by lukem12345** [comment]
Yes, we should also rethink whether the current method of passing `hodge` as a named argument to `generate` is scalable. Since there are also multiple `sharp`s now, for example.
